### DISCUSSION TITLE
How to enable "Unmanaged Attributes" via Admin CLI (kcadm.sh)

### DISCUSSION THREAD

**Discussion by RoshanGerard** [keycloak/keycloak#33859] [Archive]
Hi,

Does anyone know how to enable "Unmanaged Attributes" via Admin CLI (kcadm.sh)

I tried to enable using the below bash commands but it still indicates it is disabled in keycloak admin UI:

![image](https://github.com/user-attachments/assets/0ab73c52-886d-45a7-9fbc-8c250cb41312)

Attempt 1:
```
./kcadm.sh update realms/$KYC_REALM \
  -s attributes.unmanagedAttributesEnabled=true \
  -s attributes.unmanagedAttributePolicy="ADMIN_EDIT";
```

Attempt 2:
```
./kcadm.sh update realms/$KYC_REALM \
  -s attributes.unmanagedAttributesEnabled=true \
  -s attributes.unmanagedAttributePolicy="ENABLED";
```

I tried this on Keycloak version 24.0

I would appreciate it if anyone could assist with this.

Thanks,
Roshan

**Comment by codespearhead** [comment]
Try this:

```bash
COMPONENT_ID=$(./kcadm.sh get components -r $KYC_REALM | grep -B10 '"name": "'$COMPONENT_NAME'"' | grep '"id":' | sed 's/.*"id": "\(.*\)",/\1/')

./kcadm.sh update components/$COMPONENT_ID -r $KYC_REALM -s 'config.manage-attributes=["true"]'
./kcadm.sh update components/$COMPONENT_ID -r $KYC_REALM -s 'config.unmanagedAttributes=["mobile", "customAttribute"]'
```

**Reply by RoshanGerard** [reply]
Hi,

Thank you for the solution, I have tried what you suggested but that didn't enable the unmanaged attributes.
